src/Integrations/Integrations/LaravelQueue/LaravelQueueIntegration.php

```diff
@@ -74,16 +75,21 @@ function () {
 
                     $integration->setSpanAttributes($span, 'laravel.queue.process', 'receive', $job);
 
-                    // Create a new trace
                     $payload = $job->payload();
                     if (isset($payload['dd_headers'])) {
-                        $newTrace = start_trace_span();
-
-                        $integration->setSpanAttributes($newTrace, 'laravel.queue.process', 'receive', $job);
-
-                        $integration->extractContext($payload);
-                        $span->links[] = $newTrace->getLink();
-                        $newTrace->links[] = $span->getLink();
+                        if (dd_trace_env_config('DD_TRACE_LARAVEL_QUEUE_DISTRIBUTED_TRACING')) {
+                            $newTrace = start_trace_span();
+                            $integration->setSpanAttributes($newTrace, 'laravel.queue.process', 'receive', $job);
+                            $integration->extractContext($payload);
+                            $span->links[] = $newTrace->getLink();
+                            $newTrace->links[] = $span->getLink();
+                        } else {
+                            $ddHeaders = $payload['dd_headers'];
+                            $spanLink = new SpanLink();
+                            $spanLink->traceId = $ddHeaders[Propagator::DEFAULT_TRACE_ID_HEADER];;
+                            $spanLink->spanId = $ddHeaders[Propagator::DEFAULT_PARENT_ID_HEADER];
```

Comment:
This is a) assuming that datadog headers are distributed and b) will it discard the upper 64 bit of 128 bit headers.

I'll fix this.

Reply:
This can now be fixed using `SpanLink::fromHeaders()`, after a rebase on top of #2314.